### PR TITLE
explain using PTFs in SUSE Manager / Uyuni

### PR DESCRIPTION
**Pull request by mcalmer**
# Description

Project Lotus adds the functionality to use PTF in Uyuni. Describe how to do it.

SMEs: @mackdk @mcalmer 

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.3 when PR is merged https://github.com/SUSE/spacewalk/pull/19933
- [ ] Manager-4.2

# Links

Fixes https://github.com/SUSE/spacewalk/issues/20322

### CODE REVIEW

File: modules/administration/pages/ptfs.adoc
Comment (mackdk) [inline modules/administration/pages/ptfs.adoc:5]:
```suggestion
SUSE provides temporary fixes for all currently supported solutions shipped directly to its customers.
These PTFs (Program Temporary Fixes) are now available as repositories, which can be synced in {productname}.
```

File: modules/administration/pages/ptfs.adoc
Comment (mackdk) [inline modules/administration/pages/ptfs.adoc:11]:
```suggestion
They will depend on the correct version of the package that is known to include the software correction. This type of package:
```

File: modules/administration/pages/ptfs.adoc
Comment (mackdk) [inline modules/administration/pages/ptfs.adoc:16]:
```suggestion
* cannot be installed accidentally (i.e. zypper update will never suggest installing them),
* cannot be removed accidentally (i.e. a newer package version will not replace the PTF one, unless the user makes it explicitly on the zypper command line),
* is only updated when the newer version is known to address the specific issue previously solved by the PTF,
* updates only packages already installed on the system (i.e. if a software is split into multiple packages, the PTF will replace only those currently installed on the system).
```

File: modules/administration/pages/ptfs.adoc
Comment (admd) [inline modules/administration/pages/ptfs.adoc:4]:
```suggestion
SUSE provides for their Enterprise Products temporary fixes shipped to individual customers.
```

**Review by admd — COMMENTED** [review]
Overall looks good, just a couple of suggestion

1. Anything to be noted down about mixed env where people are installing some PTFs(e.g mass ones) in traditional way and then in some case though this new way?
2. Some people do manual stuff from terminal, would be good to add some info about `zypper removeptf `command
3. In case new MU is released with the fix, can/do we show in UI that these installed PTFs can be removed now safely or not needed?
4. Do we provide API too? if yes, we should add a note here as well.
5.  We could already add a note about syncing optional channel from UI as that part will be done till then.

File: modules/administration/pages/ptfs.adoc
Comment (mackdk) [inline modules/administration/pages/ptfs.adoc:41]:
```suggestion
PTF packages need to be installed explicitly, since they are not automatically picked up when updating a system.
The SUSE Customer Support will provide the PTF number to fix a specific problem.
In {productname} {webui} every system with PTFs available for installation has a page which list them.
```

File: modules/administration/pages/ptfs.adoc
Comment (mackdk) [inline modules/administration/pages/ptfs.adoc:46]:
```suggestion
. In the {productname} {webui}, navigate to menu:Systems[Systems List] and select the client where you want to install a PTF.
```

**Review by mackdk — COMMENTED** [review]
Looks good to me. I added some minor rewording proposal. Feel free to integrate them if you think they simplify the text.

File: modules/administration/pages/ptfs.adoc
Comment (mcalmer) [reply modules/administration/pages/ptfs.adoc]: used that. thanks

File: modules/administration/pages/ptfs.adoc
Comment (mcalmer) [reply modules/administration/pages/ptfs.adoc:16]: This is a 1:1 copy of the TID. I will let our doc team correct this or not

File: modules/administration/pages/ptfs.adoc
Comment (mcalmer) [reply modules/administration/pages/ptfs.adoc:41]: used it

File: modules/administration/pages/ptfs.adoc
Comment (mcalmer) [reply modules/administration/pages/ptfs.adoc:46]: :+1:

**Comment by mcalmer** [comment]
> Overall looks good, just a couple of suggestion
> 
> 1. Anything to be noted down about mixed env where people are installing some PTFs(e.g mass ones) in traditional way and then in some case though this new way?

No idea how to do this. The text is getting too complex and confusing. I think the customer should decide what he use.

> 2. Some people do manual stuff from terminal, would be good to add some info about `zypper removeptf `command

added

> 3. In case new MU is released with the fix, can/do we show in UI that these installed PTFs can be removed now safely or not needed?

I think this is not needed. A normal update will replace it when a MU will have a newer version. I added that to the text.

> 4. Do we provide API too? if yes, we should add a note here as well.

The normal package API can be used. I added this to the text.

> 5. We could already add a note about syncing optional channel from UI as that part will be done till then.

Yes, but I think @parlt91 should add this. Pascal: please add a section about enabling optional channels via Web UI to this draft.

File: modules/administration/pages/ptfs.adoc
Comment (keichwa) [inline modules/administration/pages/ptfs.adoc:56]:
```suggestion
In case a PTF should be installed using the API, the normal [systemitem]``system.schedulePackageInstall`` API can be used with the proxy package name.
```

File: modules/administration/pages/ptfs.adoc
Comment (keichwa) [inline modules/administration/pages/ptfs.adoc:78]:
```suggestion
. Navigate to the menu:Systems[Software > PTFs > List/Remove] sub tab.
. Select the ptf package you want to remove.
```

File: modules/administration/pages/ptfs.adoc
Comment (keichwa) [inline modules/administration/pages/ptfs.adoc:96]:
```suggestion
In case a PTF should be removed directly on the client using the console, it is required to use a special command ``zypper removeptf ...``.
  All other ways result either in an error or can lead to unwanted behavior like removing important packages from the system and make the system unusable.
```

**Review by keichwa — APPROVED** [review]
Some formal change proposals.
When it is merged we can propose more cosmetic changes according to our styleguide.

File: modules/administration/pages/ptfs.adoc
Comment (ktsamis) [inline modules/administration/pages/ptfs.adoc:10]:
```suggestion
PTF packages are installed via a proxy package and are named `ptf-xxxxxx`, where xxxxxx is a number and part of the name of the package, not its version.

```

File: modules/administration/pages/ptfs.adoc
Comment (ktsamis) [inline modules/administration/pages/ptfs.adoc:68]:
```suggestion
The replacement of the PTF with the maintenance update which includes the fix happens automatically via
```

File: modules/administration/pages/ptfs.adoc
Comment (ktsamis) [inline modules/administration/pages/ptfs.adoc:96]:
```suggestion
All other ways result either in an error or can lead to unwanted behavior like removing important packages from the system and make the system unusable.
```

**Review by ktsamis — COMMENTED** [review]
Small typos and a small clarification, hopefully it helps :)

**Comment by keichwa** [comment]
I agree with @ktsamis suggestions and fixes, thanks!

**Review by keichwa — APPROVED** [review]
Before merging we probably must run the l10n-weblate script because here we add a new file.  @jcayouette

**Comment by mcalmer** [comment]
Removing "Draft" as now also the doc from Pascal is in about adding PTFs repos via UI.

**Pull request merged**
